### DISCUSSION TITLE
Example of how to add a login page?

### DISCUSSION THREAD

**Discussion by thisismygitrepo** [vanna-ai/vanna#376] [General]
I've never done Flask webapp, can someone please add an example of how to hit users with login page first?

at the momment, I jump to `app.run()` as per examples.

Thanks

**Comment by zainhoda** [comment]
Have you seen the example here by any chance?

https://vanna.ai/docs/web-app-auth/#example

**Comment by thisismygitrepo** [comment]
Thanks for pointing out the example. I just compared to what I did with help from AI and I believe that the interface (template) provided is positively misleading. For instance, it says that many methods should return `str` where as the working example you provided returns a response object `Response`.

I might need to raise an issue as the type checker is yelling at me for violating the ABC template class when I use the example that is.
